src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -513,11 +513,25 @@ private QuantityValidationResponse exactStringMatch(final IsaacNumericQuestion i
                                                         final Quantity answerFromUser) {
 
         log.debug("\t[exactStringMatch]");
+        boolean wasACorrectAnswerWithUsersSelectedUnit = false;
```

Comment:
Is this variable necessary? Or, rather, since this is necessary, why is `unitsCorrect` still necessary? See the comment further down about the `unitsCorrect` variable.